src/modules/qt/gps_parser.cpp

```diff
@@ -686,71 +773,68 @@ void xml_parse_gpx(xmlNodeSetPtr found_nodes, gps_point_ll **gps_list, int *coun
-			xmlFree(str);
-		}
-		if (xmlHasProp(crt_node, _x("lon"))) {
-			xmlChar* str = xmlGetProp(crt_node, _x("lon"));
-			crt_point.lon = strtod(_s(str), NULL);
-			xmlFree(str);
-		}
-		for (gps_val = crt_node->children; gps_val; gps_val = gps_val->next)
+		reader.readNext();
+		// mlt_log_info(NULL, " readNext tag: %s type = %d\n", _qxml_getthestring(reader.name()), reader.tokenType());
+		if (reader.isStartElement() && reader.name() == "trkpt")
 		{
-			if (strncmp(_s(gps_val->name), "ele", strlen("ele")) == 0)
-				crt_point.ele = strtod(_s(gps_val->children->content), NULL);
-			else if (strncmp(_s(gps_val->name), "time", strlen("time")) == 0)
-				crt_point.time = datetimeXMLstring_to_mseconds(_s(gps_val->children->content), NULL);
-			else if (strncmp(_s(gps_val->name), "bearing", strlen("bearing")) == 0)
-				crt_point.bearing = (int)strtod(_s(gps_val->children->content), NULL);
-			else if (strncmp(_s(gps_val->name), "speed", strlen("speed")) == 0)
-				crt_point.speed = strtod(_s(gps_val->children->content), NULL);
-			else if (strncmp(_s(gps_val->name), "extensions", strlen("extensions")) == 0)
+			gps_point_raw crt_point = uninit_gps_raw_point;
+			
+			QXmlStreamAttributes attributes = reader.attributes();
+			if (attributes.hasAttribute("lat"))
+				crt_point.lat = attributes.value("lat").toDouble();
+			if (attributes.hasAttribute("lon"))
+				crt_point.lon = attributes.value("lon").toDouble();
+			
+			while (reader.readNext() && !(reader.name() == "trkpt" && reader.tokenType() == QXmlStreamReader::EndElement)) //until closing trkpt
 			{
-				for (gps_subval = gps_val->children; gps_subval; gps_subval=gps_subval->next) {
-					if (strncmp(_s(gps_subval->name), "gpxtpx:TrackPointExtension", strlen("gpxtpx:TrackPointExtension")) == 0) {
-						for (gps_subsubval = gps_subval->children; gps_subsubval; gps_subsubval=gps_subsubval->next) {
-							if (strncmp(_s(gps_subsubval->name), "gpxtpx:hr", strlen("gpxtpx:hr")) == 0) {
-								crt_point.hr = (short)strtod(_s(gps_subsubval->children->content), NULL);
-							}
-						}
+				// mlt_log_info(NULL, "[trkpt->1] readNext tag: %s, type=%d \n", _qxml_getthestring(reader.name()), reader.tokenType());
+				if (!reader.isStartElement())
+					continue;
+
+				if (reader.name() == "ele")
+					crt_point.ele = reader.readElementText().toDouble();
+				else if (reader.name() == "time")
+					crt_point.time = datetimeXMLstring_to_mseconds(qUtf8Printable(reader.readElementText()));
+				else if (reader.name() == "speed")
+					crt_point.speed = reader.readElementText().toDouble();
+				else if (reader.name() == "course")
+					crt_point.bearing = reader.readElementText().toDouble();
+				else if (reader.name() == "extensions")
+				{
+					reader.readNextStartElement();
+					if (reader.name() == "TrackPointExtension")
+					{
+						reader.readNextStartElement();
+						if (reader.name() == "hr")
+							crt_point.hr = reader.readElementText().toDouble();
```

Comment:
All of these name comparisons fail to compile on Qt 6:
`../src/modules/qt/gps_parser.cpp:796:26: error: conversion from ‘const char [4]’ to ‘QChar’ is ambiguous`